server/src/test/java/org/opensearch/cluster/routing/allocation/RemoteStoreMigrationAllocationDeciderTests.java

```diff
@@ -0,0 +1,964 @@
+                        )
+                            .addShard(
+                                // new replica's allocation
+                                TestShardRouting.newShardRouting(
+                                    shardId.getIndexName(),
+                                    shardId.getId(),
+                                    null,
+                                    false,
+                                    ShardRoutingState.UNASSIGNED
+                                )
+                            )
+                            .build()
+                    )
+            )
+            .build();
+
+        DiscoveryNodes discoveryNodes = DiscoveryNodes.builder().add(remoteNode).localNodeId(remoteNode.getId()).build();
+
+        ClusterState clusterState = ClusterState.builder(ClusterName.DEFAULT)
+            .metadata(metadata)
+            .routingTable(routingTable)
+            .nodes(discoveryNodes)
+            .build();
+
+        assertEquals(2, clusterState.getRoutingTable().allShards().size());
+
+        ShardRouting replicaShardRouting = clusterState.getRoutingTable().shardRoutingTable(TEST_INDEX, 0).replicaShards().get(0);
+        RoutingNode remoteRoutingNode = clusterState.getRoutingNodes().node(remoteNode.getId());
+
+        RemoteStoreMigrationAllocationDecider remoteStoreMigrationAllocationDecider = new RemoteStoreMigrationAllocationDecider(
+            customSettings,
+            getClusterSettings(customSettings)
+        );
+
+        RoutingAllocation routingAllocation = new RoutingAllocation(
+            new AllocationDeciders(Collections.singleton(remoteStoreMigrationAllocationDecider)),
+            clusterState.getRoutingNodes(),
+            clusterState,
+            null,
+            null,
+            0L
+        );
+        routingAllocation.debugDecision(true);
+
+        Decision decision = remoteStoreMigrationAllocationDecider.canAllocate(replicaShardRouting, remoteRoutingNode, routingAllocation);
+        assertThat(decision.type(), is(Decision.Type.NO));
+        assertThat(
+            decision.getExplanation().toLowerCase(Locale.ROOT),
+            is(
+                "[remote_store migration_direction]: replica shard copy can not be allocated to a remote node since primary shard for this replica is not yet active"
+            )
+        );
+    }
+
+    // prepare index metadata for test-index
+    private IndexMetadata.Builder getIndexMetadataBuilder(boolean isRemoteStoreBackedIndex, int shardCount, int replicaCount) {
+        return IndexMetadata.builder(TEST_INDEX)
+            .settings(
+                settings(Version.CURRENT).put(SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
+                    .put(SETTING_REMOTE_SEGMENT_STORE_REPOSITORY, TEST_REPO)
+                    .put(SETTING_REMOTE_TRANSLOG_STORE_REPOSITORY, TEST_REPO)
+                    .put(SETTING_REMOTE_STORE_ENABLED, isRemoteStoreBackedIndex)
+                    .build()
```

Comment:
Lets only put `repository and `replication type` information when it is remote enabled .